File: aihound/scanners/claude_desktop.py
"""Scanner for Claude Desktop application credentials."""

from __future__ import annotations

from pathlib import Path

from aihound.core.scanner import BaseScanner, ScanResult
from aihound.core.platform import (
    detect_platform, Platform, get_home, get_appdata, get_wsl_windows_home,
)
from aihound.core.mcp import parse_mcp_file
from aihound.scanners import register


@register
class ClaudeDesktopScanner(BaseScanner):
    def name(self) -> str:
        return "Claude Desktop"

    def slug(self) -> str:
        return "claude-desktop"

    def scan(self, show_secrets: bool = False) -> ScanResult:
        plat = detect_platform()
        result = ScanResult(scanner_name=self.name(), platform=plat.value)

        for path in self._get_config_paths(plat):
            findings, errors = parse_mcp_file(path, self.name(), show_secrets)
            result.findings.extend(findings)
            result.errors.extend(errors)

        return result

    def _get_config_paths(self, plat: Platform) -> list[Path]:
        paths = []

        if plat == Platform.MACOS:
            paths.append(
                get_home() / "Library" / "Application Support" / "Claude" / "claude_desktop_config.json"
            )

        elif plat == Platform.WINDOWS:
            appdata = get_appdata()
            if appdata:
                paths.append(appdata / "Claude" / "claude_desktop_config.json")

        elif plat == Platform.LINUX:
            from aihound.core.platform import get_xdg_config
            paths.append(get_xdg_config() / "Claude" / "claude_desktop_config.json")

        elif plat == Platform.WSL:
            # Linux-side
            from aihound.core.platform import get_xdg_config
            paths.append(get_xdg_config() / "Claude" / "claude_desktop_config.json")
            # Windows-side
            appdata = get_appdata()
            if appdata:
                paths.append(appdata / "Claude" / "claude_desktop_config.json")

        return paths

File: Other Versions/Go/scanners/claude_desktop.go
package scanners

import (
	"path/filepath"

	"aihound/core"
)

type claudeDesktopScanner struct{}

func init() {
	Register(&claudeDesktopScanner{})
}

func (s *claudeDesktopScanner) Name() string      { return "Claude Desktop" }
func (s *claudeDesktopScanner) Slug() string       { return "claude-desktop" }
func (s *claudeDesktopScanner) IsApplicable() bool { return true }

func (s *claudeDesktopScanner) Scan(showSecrets bool) core.ScanResult {
	result := core.ScanResult{ScannerName: s.Name(), Platform: core.DetectPlatform().String()}
	plat := core.DetectPlatform()

	for _, p := range s.getConfigPaths(plat) {
		findings := core.ParseMCPFile(p, s.Name(), showSecrets)
		result.Findings = append(result.Findings, findings...)
	}

	return result
}

func (s *claudeDesktopScanner) getConfigPaths(plat core.Platform) []string {
	var paths []string

	switch plat {
	case core.PlatformMacOS:
		paths = append(paths,
			filepath.Join(core.GetHome(), "Library", "Application Support", "Claude", "claude_desktop_config.json"),
		)

	case core.PlatformWindows:
		if appdata := core.GetAppData(); appdata != "" {
			paths = append(paths,
				filepath.Join(appdata, "Claude", "claude_desktop_config.json"),
			)
		}

	case core.PlatformLinux:
		paths = append(paths,
			filepath.Join(core.GetXDGConfig(), "Claude", "claude_desktop_config.json"),
		)

	case core.PlatformWSL:
		// Linux-side
		paths = append(paths,
			filepath.Join(core.GetXDGConfig(), "Claude", "claude_desktop_config.json"),
		)
		// Windows-side
		if appdata := core.GetAppData(); appdata != "" {
			paths = append(paths,
				filepath.Join(appdata, "Claude", "claude_desktop_config.json"),
			)
		}
	}

	return paths
}
